### PR TITLE
Add support for TS config files

### PR DESCRIPTION
**Pull request by itsyoboieltr**
## Description

This PR adds prettier.config.ts support, addressing: https://github.com/prettier/prettier/issues/16119

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Add flag here should be able to test https://github.com/prettier/prettier/blob/2c8440ebeebf54833ba0cfb0eb5df8e9939e667a/tests/integration/run-cli.js#L40

File: website/versioned_docs/version-stable/configuration.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ You can configure Prettier via (in order of precedence):
...
 - A `.prettierrc` file written in JSON or YAML.
 - A `.prettierrc.json`, `.prettierrc.yml`, `.prettierrc.yaml`, or `.prettierrc.json5` file.
-- A `.prettierrc.js`, or `prettier.config.js` file that exports an object using `export default` or `module.exports` (depends on the [`type`](https://nodejs.org/api/packages.html#type) value in your `package.json`).
-- A `.prettierrc.mjs`, or `prettier.config.mjs` file that exports an object using `export default`.
-- A `.prettierrc.cjs`, or `prettier.config.cjs` file that exports an object using `module.exports`.
+- A `.prettierrc.js`, `.prettierrc.ts`, `prettier.config.js`, or `prettier.config.ts` file that exports an object using `export default` or `module.exports` (depends on the [`type`](https://nodejs.org/api/packages.html#type) value in your `package.json`).
+- A `.prettierrc.mjs`, `.prettierrc.mts`, `prettier.config.mjs`, or `prettier.config.mts` file that exports an object using `export default`.
+- A `.prettierrc.cjs`, `.prettierrc.cts`, `prettier.config.cjs`, or `prettier.config.cts` file that exports an object using `module.exports`.
 - A `.prettierrc.toml` file.
```
Comment (fisker) [inline website/versioned_docs/version-stable/configuration.md:15]: This file will be updated when release new version, please revert.

File: tests/format/markdown/markdown/real-world-case.md [hunk 2]
```diff
@@ -702,8 +702,8 @@ Default | CLI Override | API Override
 Prettier uses [cosmiconfig](https://github.com/davidtheclark/cosmiconfig) for configuration file support.
 This means you can configure prettier via:
 
-* A `.prettierrc` file, written in YAML or JSON, with optional extensions: `.yaml/.yml/.json/.js`.
-* A `prettier.config.js` file that exports an object.
+* A `.prettierrc` file, written in YAML or JSON, with optional extensions: `.yaml/.yml/.json/.js/.ts`.
+* A `prettier.config.js` or `prettier.config.ts` file that exports an object.
 * A `"prettier"` key in your `package.json` file.
```
Comment (fisker) [inline tests/format/markdown/markdown/real-world-case.md:707]: This file is for markdown format test, please revert.

File: tests/integration/run-cli.js [hunk 3]
```diff
@@ -37,7 +37,11 @@ function runCliWorker(dir, args, options) {
 
   const worker = new Worker(CLI_WORKER_FILE, {
     argv: args,
-    execArgv: ["--trace-deprecation"],
+    execArgv: [
+      "--trace-deprecation",
+      "--disable-warning=ExperimentalWarning",
+      "--experimental-strip-types",
```
Comment (kachkaev) [inline tests/integration/run-cli.js:43]: Would this work in Node < 22.6 where this [flag was introduced](https://nodejs.org/en/blog/release/v22.6.0)? We should make sure we don’t get any warnings in all supported Node versions and if the flag is not available, there should be a meaningful error about why `prettier.config.ts` could not load. Docs need to reflect min version for TS config too, unless we want to change the approach and use a built-in transpiler.

Comment (fisker) [reply tests/integration/run-cli.js:43]: It should be passed via `options` and only set when testing ts configs.

**Review by fisker — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**